### PR TITLE
[READY]PAI PATCH 2

### PR DESCRIPTION
**Pull request by silicons**
:cl: Mekhi Anderson
rscdel: Fixes various PAI bugs, various tweaks and bullshit.
/:cl:

### CODE REVIEW

**Comment by Supermichael777** [comment]
make their grabs return

fucking shitlers keep trying top make sec play with the slip and slide and taking prizoners on mr pais wild ride

**Comment by kevinz000** [comment]
kor is doing that, pais will be out untill everything gets fixed

**Comment by TechnoAlchemisto** [comment]
foolish

**Comment by kilkun** [comment]
be nice to pais

**Comment by Grazyn** [comment]
Is there a button so the owner of the pAI (or whoever holds it) can disable holo projection? Because there should be.

File: code/modules/mob/living/silicon/pai/life.dm [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-/mob/living/silicon/pai/updatehealth()
-	if(status_flags & GODMODE)
-		return
-	health = maxHealth - getBruteLoss() - getFireLoss()
-	update_stat()
+/mob/living/silicon/pai/Life()
+	if(stat == DEAD)
+		return
+	if(cable)
+		if(get_dist(src, src.cable) > 1)
```
Comment (AnturK) [inline code/modules/mob/living/silicon/pai/life.dm]: Remove redundant src. please.

File: code/modules/mob/living/silicon/pai/life.dm [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+			var/turf/T = get_turf(src.loc)
+			T.visible_message("<span class='warning'>[src.cable] rapidly retracts back into its spool.</span>", "<span class='italics'>You hear a click and the sound of wire spooling rapidly.</span>")
+			qdel(src.cable)
+			cable = null
+	silent = max(silent - 1, 0)
+	if(weakened > 0)
```
Comment (AnturK) [inline code/modules/mob/living/silicon/pai/life.dm]: Same function as above, different code, keep it consistent please.

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 3]
```diff
@@ -4,7 +4,7 @@
 	var/obj/machinery/camera/current = null
 	icon = 'icons/mob/pai.dmi'
 	icon_state = "repairbot"
-	mouse_opacity = 1
+	mouse_opacity = 2
```
Comment (AnturK) [inline code/modules/mob/living/silicon/pai/pai.dm]: Why ? Fill the sprite with 0.0.0.1 zones if it's hard to click.

File: code/modules/mob/living/silicon/pai/say.dm [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /mob/living/silicon/pai/say(msg)
-	if(silence_time)
+	if(silent > 0)
```
Comment (AnturK) [inline code/modules/mob/living/silicon/pai/say.dm]: Can be just if(silent)

File: code/modules/mob/living/silicon/pai/software.dm [hunk 5]
```diff
@@ -211,7 +211,7 @@
 				else if(href_list["ringer"])
 					pda.silent = !pda.silent
 				else if(href_list["target"])
-					if(silence_time)
+					if(silent > 0)
```
Comment (AnturK) [inline code/modules/mob/living/silicon/pai/software.dm]: Same as above

**Review by AnturK — CHANGES_REQUESTED** [review]
Some minor issues.

**Comment by TehZombehz** [comment]
Can you not just say it 'fixes numerous pAI bugs'? Then again, I dunno if we prefer longer or shorter changelogs.

**Comment by AnturK** [comment]
@kevinz000 Please, this is just minor touchups.

File: code/modules/mob/living/silicon/pai/pai.dm (hunk 3 above)
Comment (kevinz000) [inline code/modules/mob/living/silicon/pai/pai.dm]: That's how bees worked and I don't know how to do that?

**Comment by kevinz000** [comment]
WIP pending damage refactor for PAIs.

File: code/modules/mob/living/silicon/pai/pai.dm (hunk 3 above)
Comment (optimumtact) [inline code/modules/mob/living/silicon/pai/pai.dm]: @AnturK we have this system for this exact reason, so you don't have to do that.

Comment (AnturK) [inline code/modules/mob/living/silicon/pai/pai.dm]: This makes clicking anything that's not pAI impossible. I guess it's not a big difference but it feels lazy.

**Review by AnturK — APPROVED** [review]

File: code/modules/mob/living/silicon/pai/pai.dm (hunk 3 above)
Comment (RemieRichards) [inline code/modules/mob/living/silicon/pai/pai.dm]: @AnturK don't fill it with alpha 1 white pixels, when I asked WJ for the bee sprites, he pre-emptively filled the edges with alpha 1 white pixels, and I could see them in game and it was horrible, which is why I removed those pixels and used ```mouse_opacity```

File: code/modules/mob/living/life.dm [hunk 6]
```diff
@@ -46,11 +33,12 @@
 		machine.check_eye(src)
 
 	if(stat != DEAD)
-		handle_disabilities() // eye, ear, brain damages
-	if(stat != DEAD)
-		handle_status_effects() //all special effects, stunned, weakened, jitteryness, hallucination, sleeping, etc
-
-	if(stat != DEAD)
+		handle_disabilities()	// eye, ear, brain damages
```
Comment (kevinz000) [inline code/modules/mob/living/life.dm]: Erm.

**Comment by kevinz000** [comment]
@optimumtact nothing bad should happen, did a quick check.

**Comment by kevinz000** [comment]
ok, reverted life because i dont' want to get blamed if shit happens again...

**Pull request merged**